src/PerformanceSensitiveAnalyzers/CSharp/CallSiteImplicitAllocationAnalyzer.cs

```diff
@@ -42,65 +40,33 @@ internal sealed class CallSiteImplicitAllocationAnalyzer : AbstractAllocationAna
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ParamsParameterRule, ValueTypeNonOverridenCallRule);
 
-        protected override ImmutableArray<SyntaxKind> Expressions => ImmutableArray.Create(SyntaxKind.InvocationExpression);
+        protected override ImmutableArray<OperationKind> Operations => ImmutableArray.Create(OperationKind.Invocation);
 
         private static readonly object[] EmptyMessageArgs = Array.Empty<object>();
 
-        protected override void AnalyzeNode(SyntaxNodeAnalysisContext context, in PerformanceSensitiveInfo info)
+        protected override void AnalyzeNode(OperationAnalysisContext context, in PerformanceSensitiveInfo info)
         {
-            var node = context.Node;
-            var semanticModel = context.SemanticModel;
-            Action<Diagnostic> reportDiagnostic = context.ReportDiagnostic;
-            var cancellationToken = context.CancellationToken;
-
-            var invocationExpression = node as InvocationExpressionSyntax;
-
-            if (semanticModel.GetSymbolInfo(invocationExpression, cancellationToken).Symbol is IMethodSymbol methodInfo)
+            if (!(context.Operation is IInvocationOperation invocation))
```

Comment:
I believe you can cast the operation directly. The registered operation above are only OperationKind.Invocation, so it seems impossible for context.Operation not to be IInvocationOperation.